app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomeDestination.kt

```diff
@@ -229,32 +229,40 @@ private fun HomeScreen(
 
     Column {
       TopAppBarLayoutForActions {
-        for (action in uiState.topBarActions) {
-          when (action) {
-            HomeTopBarAction.ChatAction -> ToolbarChatIcon(
-              onClick = onStartChat,
-              modifier = Modifier.notificationCircle(uiState.hasUnseenChatMessages),
-            )
-
-            is HomeTopBarAction.CrossSellsAction -> ToolbarCrossSellsIcon(
-              onClick = {
-                crossSellsForBottomSheet = action.crossSells
-              },
-            )
+        val currentState = uiState as? HomeUiState.Success
+        if (currentState != null) {
+          val actionsList = buildList {
+            if (currentState.crossSellsAction != null) add(currentState.crossSellsAction)
+            if (currentState.firstVetAction != null) add(currentState.firstVetAction)
+            if (currentState.chatAction != null) add(currentState.chatAction)
+          }
+          for (action in actionsList) {
+            when (action) {
+              HomeTopBarAction.ChatAction -> ToolbarChatIcon(
+                onClick = onStartChat,
+                modifier = Modifier.notificationCircle(uiState.hasUnseenChatMessages),
+              )
 
-            is HomeTopBarAction.FirstVetAction -> {
-              val sections = action.sections
-              ToolbarFirstVetIcon(
-                onClick = { navigateToFirstVet(sections) },
+              is HomeTopBarAction.CrossSellsAction -> ToolbarCrossSellsIcon(
+                onClick = {
+                  crossSellsForBottomSheet = action.crossSells
+                },
               )
+
+              is HomeTopBarAction.FirstVetAction -> {
+                val sections = action.sections
+                ToolbarFirstVetIcon(
+                  onClick = { navigateToFirstVet(sections) },
+                )
+              }
+            }
+            if (actionsList.indexOf(action) != actionsList.lastIndex) {
+              Spacer(modifier = Modifier.width(8.dp))
             }
```

Comment:
I've found it more convenient to do this check the other way around lately, by doing what is done here https://github.com/HedvigInsurance/android/blob/0ec4aaee4da0c6552868e06c462ab30c2536de5b/app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments/discounts/DiscountRow.kt#L39 of iterating over the list with an index, and if the index is not 0 to add the "in-between" item then.
When doing it on a for-loop you can do it this way https://github.com/HedvigInsurance/android/blob/638e6dc11eacef74ff4ed95815805c87d7d498a0/app/feature/feature-help-center/src/main/kotlin/com/hedvig/android/feature/help/center/ui/HelpCenterSectionWithClickableRows.kt#L43 (ignore the `> 0` there, I think `!= 0` is more readable), so the withIndex() function. The IDE actually helps you by suggesting it if you do Option Enter while hovering on the `in` keyword in the `for ... in ...` I think.